electrum/lnpeer.py

```diff
@@ -220,12 +220,100 @@ def process_message(self, message):
             if asyncio.iscoroutinefunction(f):
                 asyncio.ensure_future(self.taskgroup.spawn(execution_result))
 
-    def on_error(self, payload):
+    def on_warning(self, payload):
+        """When we receive a warning, we put it into the ordered message queue for the channel.
+        If we `self.wait_for_message` in a protocol action, this will lead to a disconnect via
+        an exception raised therein.
+        """
+        # TODO: we could need some reconnection logic here -> delayed reconnect
+        self.logger.info(f"remote peer sent warning [DO NOT TRUST THIS MESSAGE]: {payload['data'].decode('ascii')}")
+        channel_id = payload.get("channel_id")
+        if channel_id == bytes(32):
+            for channel_id in self.channels.keys():
+                self.ordered_message_queues[channel_id].put_nowait((None, {'warning': payload['data']}))
+        # we only put into the message queue if we know about the channel
+        elif channel_id in self.temp_id_to_id:
+            channel_id = self.temp_id_to_id[channel_id]
+        elif channel_id in self.channels:
+            pass
+        else:
+            channel_id = None
+        if channel_id is not None:
+            self.ordered_message_queues[channel_id].put_nowait((None, {'warning': payload['data']}))
```

Comment:
What if `channel_id` was all zeroes at the beginning, i.e. line 231 evaluated to True?
The branching and logic here is a bit confusing...
L232 runs, and the loop rebinds `channel_id`, so now it will have the value of one of the channels we actually have,
and then L241 is True, and L242 runs, and here we put the same warning again for that channel?

Same for `on_error`.

Reply:
I tried to improve it.